storm-core/src/jvm/org/apache/storm/scheduler/blacklist/strategies/DefaultBlacklistStrategy.java

```diff
@@ -0,0 +1,142 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ * <p/>
+ * http://www.apache.org/licenses/LICENSE-2.0
+ * <p/>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.storm.scheduler.blacklist.strategies;
+
+import org.apache.storm.scheduler.*;
+import org.apache.storm.scheduler.blacklist.CircularBuffer;
+import org.apache.storm.scheduler.blacklist.reporters.IReporter;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.*;
+
+public class DefaultBlacklistStrategy implements IBlacklistStrategy {
+
+    private static Logger LOG = LoggerFactory.getLogger(DefaultBlacklistStrategy.class);
+
+    private IReporter _reporter;
+
+    private int _toleranceTime;
+    private int _toleranceCount;
+    private int _resumeTime;
+    private int _nimbusMonitorFreqSecs;
+
+    private TreeMap<String, Integer> blacklist;
+
+    @Override
+    public void prepare(IReporter reporter, int toleranceTime, int toleranceCount, int resumeTime, int nimbusMonitorFreqSecs) {
+        _reporter = reporter;
+
+        _toleranceTime = toleranceTime;
+        _toleranceCount = toleranceCount;
+        _resumeTime = resumeTime;
+        _nimbusMonitorFreqSecs = nimbusMonitorFreqSecs;
+        blacklist = new TreeMap<>();
+    }
+
+    @Override
+    public Set<String> getBlacklist(CircularBuffer<HashMap<String, Set<Integer>>> toleranceBuffer, Cluster cluster, Topologies topologies) {
```

Comment:
The only thing that's ever actually used in `toleranceBuffer` here (except for logging it) is the key of the hashmap. It looks like `toleranceBuffer` should really be a `List<String>` called `supervisorsWithFailures` or something like that. Currently, both the type and the name are misleading.